src/prefect/_internal/concurrency/from_async.py

```diff
@@ -0,0 +1,65 @@
+import asyncio
+import threading
+
+from prefect._internal.concurrency.futures import (
```

Comment:
I very much appreciate the two-module structure and the separate `futures` module